website/docs/r/policy_host_transport_node_profile.html.markdown

```diff
@@ -0,0 +1,120 @@
+---
+subcategory: "Fabric"
+layout: "nsxt"
+page_title: "NSXT: nsxt_policy_host_transport_node_profile"
+description: A resource to configure a Policy Host Transport Node Profile.
+---
+
+# nsxt_transport_node
+
+This resource provides a method for the management of a Policy Host Transport Node Profile.
+This resource is supported with NSX 4.1.0 onwards.
+
+## Example Usage
+```hcl
+resource "nsxt_policy_host_transport_node_profile" "test" {
+  display_name = "test_policy_host_tnp"
+
+  standard_host_switch {
+    host_switch_mode = "STANDARD"
+    host_switch_type = "NVDS"
+    ip_assignment {
+      assigned_by_dhcp = true
+    }
+    transport_zone_endpoint {
+      transport_zone_id = data.nsxt_transport_zone.tz1.id
+    }
+    host_switch_profile_id = [nsxt_uplink_host_switch_profile.hsw_profile1.id]
+    is_migrate_pnics       = false
+    pnics {
+      device_name = "fp-eth0"
+      uplink_name = "uplink1"
+    }
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `display_name` - (Required) Display name of the resource.
+* `description` - (Optional) Description of the resource.
+* `tag` - (Optional) A list of scope + tag pairs to associate with this resource.
+* `ignore_overridden_hosts` - (Optional) Determines if cluster-level configuration should be applied on overridden hosts
+* `standard_host_switch` - (Optional) Standard host switch specification.
+    * `cpu_config` - (Optional) Enhanced Networking Stack enabled HostSwitch CPU configuration.
+        * `num_lcores` - (Required) Number of Logical cpu cores (Lcores) to be placed on a specified NUMA node.
+        * `numa_node_index` - (Required) Unique index of the Non Uniform Memory Access (NUMA) node.
+    * `host_switch_id` - (Optional) The host switch id. This ID will be used to reference a host switch.
+    * `host_switch_mode` - (Optional) Operational mode of a HostSwitch. Accepted values - 'STANDARD', 'ENS', 'ENS_INTERRUPT' or 'LEGACY'. The default value is 'STANDARD'.
+    * `host_switch_profile_id` - (Optional) Identifiers of host switch profiles to be associated with this host switch.
+    * `host_switch_type` - (Optional) Type of HostSwitch. Accepted values - 'NVDS' or 'VDS'. The default value is 'NVDS'.
+    * `ip_assignment` - (Required) - Specification for IPs to be used with host switch virtual tunnel endpoints. Should contain exatly one of the below:
+        * `assigned_by_dhcp` - (Optional) Enables DHCP assignment. Should be set to true.
```

Comment:
I don't get the `Should be set to true` part :)

Reply:
[IpAssignmentSpec](https://vdc-download.vmware.com/vmwb-repository/dcr-public/201a09cb-5610-49e6-9d31-7db0920e3a65/65f199d6-8516-4092-9b19-906559f972b5/api_includes/types_IpAssignmentSpec.html) is a polymorphic schema with StaticIpListSpec, StaticIpPoolSpec containing members. Instead [AssignedByDhcp](https://vdc-download.vmware.com/vmwb-repository/dcr-public/201a09cb-5610-49e6-9d31-7db0920e3a65/65f199d6-8516-4092-9b19-906559f972b5/api_includes/types_AssignedByDhcp.html) has no members - it's a struct which has only resource_type. As we can't have a schema without any attributes in TF, I added a boolean. I'd appreciate a better suggestion really as it does look weird.

Reply:
From terraform perspective polymorphic structure is hidden, but the flag should be only set to true if user wants `dhcp` feature. I would suggest to add `false` default for this flag, and explain in docs, that `static_ip` and `static_ip_mac` can only be specified if `dhcp` is set to false.

Reply:
Cool I'll do this in a separate PR as it affects transport node as well.

Reply:
Only one of assigned_by_dhcp/static_ip/static_mac/static_ip_pool can be used. It's enforced [here](https://github.com/vmware/terraform-provider-nsxt/blob/master/nsxt/resource_nsxt_transport_node.go#L1360) as I couldn't use TF ExactlyOneOf capability. That's also documented.
So if I set a false default value - it'll always add assigned_by_dhcp which will create a mess IMO.
I think that I'll leave it as is - if it is set to `false` the transaction fails (as some IP assignment should be chosen and if assigned_by_dhcp is false, then there's no other assignment enabled).
And remove "should be set to true" from doc.